github-actions/unified-status-check/src/main.ts

```diff
@@ -0,0 +1,202 @@
+import * as core from '@actions/core';
+import {context} from '@actions/github';
+import {types as graphqlTypes, params, optional, onUnion, query} from 'typed-graphqlify';
+import {Octokit} from '@octokit/rest';
+import {getAuthTokenFor, ANGULAR_ROBOT, revokeActiveInstallationToken} from '../../utils.js';
+import {CheckConclusionState, PullRequestState, StatusState} from '@octokit/graphql-schema';
+
+const unifiedStatusCheckName = 'Unified Status';
+
+async function main() {
+  /** A Github API instance. */
+  const github = new Octokit({auth: await getAuthTokenFor(ANGULAR_ROBOT)});
+
+  try {
+    /** Status matchers to ignore within the context of the action, always ignoring the actions own status. */
+    const ignored = [
+      unifiedStatusCheckName,
+      ...core.getMultilineInput('ignored', {trimWhitespace: true}),
+    ];
+    /** Status matchers which must match at least one of the current statuses . */
+    const required = core.getMultilineInput('required', {trimWhitespace: true});
+    /** The pull request triggering the event */
+    const pullRequest = (
+      await github.graphql<typeof PR_SCHEMA>(query(PR_SCHEMA).toString(), {
+        ...context.issue,
+      })
+    ).repository.pullRequest;
+    /** The status check rollup results. */
+    const statusChecks = pullRequest.commits.nodes[0].commit.statusCheckRollup;
+
+    const setStatus = async (
+      state: 'pending' | 'error' | 'failure' | 'success',
+      description?: string,
+    ) => {
+      await github.repos.createCommitStatus({
+        ...context.repo,
+        sha: pullRequest.commits.nodes[0].commit.oid,
+        context: unifiedStatusCheckName,
+        state,
+        description,
+      });
+      return;
+    };
+
+    /** If no status checks are present, or if the pull request is in a draft state the unified status is in a pending state. */
+    if (!statusChecks || pullRequest.isDraft) {
+      await setStatus('pending', 'Unified Status checks have not yet started');
+      return;
+    }
+
+    const statuses: {state: StatusState; name: string}[] = statusChecks.contexts.nodes
+      .map((checkOrStatus) => {
+        if (checkOrStatus.__typename === 'StatusContext') {
+          return {
+            state: checkOrStatus.state,
+            name: checkOrStatus.context,
+          };
+        }
+        if (checkOrStatus.__typename === 'CheckRun') {
+          return {
+            state: checkOrStatus.conclusion
+              ? checkConclusionStateToStatusStateMap.get(checkOrStatus.conclusion)!
+              : 'PENDING',
+            name: checkOrStatus.name || 'Unknown Check Run',
```

Comment:
Are the status names with spaces allowed (if real ones from GitHub). Maybe use `<unknown-check-run>` or something? (not seeing any issue with this currently- but it would be more consistent then)